storage_devices/lvm.py

```diff
@@ -0,0 +1,544 @@
+    __lock = threading.Lock()
+
+    def __init__(self, name: str = None):
+        self.name = name
+
+    def __eq__(self, other):
+        try:
+            return self.name == other.name
+        except AttributeError:
+            return False
+
+    @classmethod
+    def __get_vg_name(cls, prefix: str = "vg"):
+        with cls.__lock:
+            cls.__unique_vg_id += 1
+            return f"{prefix}{cls.__unique_vg_id}"
+
+    @staticmethod
+    def get_all_volume_groups():
+        output_lines = TestRun.executor.run(f"pvscan").stdout.splitlines()
+
+        volume_groups = {}
+        for line in output_lines:
+            if "PV" not in line:
+                continue
+
+            line_elements = line.split()
+            pv = line_elements[line_elements.index("PV") + 1]
+            vg = ""
+            if "VG" in line:
+                vg = line_elements[line_elements.index("VG") + 1]
+
+            if vg not in volume_groups:
+                volume_groups[vg] = []
+            volume_groups[vg].append(pv)
+
+        return volume_groups
+
+    @staticmethod
+    def create_vg(vg_name: str, device_paths: str):
+        if not vg_name:
+            raise ValueError("Name needed for VG creation.")
+        if not device_paths:
+            raise ValueError("Device paths needed for VG creation.")
+
+        cmd = f"vgcreate --yes {vg_name} {device_paths} "
+        TestRun.executor.run_expect_success(cmd)
+
+    @classmethod
+    def is_vg_already_present(cls, dev_number: int, device_paths: str = None):
+        if not device_paths:
+            TestRun.LOGGER.exception("No devices provided.")
+
+        volume_groups = cls.get_all_volume_groups()
+
+        for vg in volume_groups:
+            for pv in volume_groups[vg]:
+                if len(volume_groups[vg]) == dev_number and pv in device_paths:
+                    return cls(vg)
+
+        for vg in volume_groups:
+            for pv in volume_groups[vg]:
+                if pv in device_paths:
+                    TestRun.LOGGER.warning(f"Some devices are used in other LVM volume group")
+        return False
+
+    @classmethod
+    def create(cls, device_paths: str = None):
+        vg_name = cls.__get_vg_name()
+
+        VolumeGroup.create_vg(vg_name, device_paths)
+
+        volume_groups = VolumeGroup.get_all_volume_groups()
+
+        if vg_name in volume_groups:
+            return cls(vg_name)
+        else:
+            TestRun.LOGGER.error("Had not found newly created VG.")
+
+    @staticmethod
+    def remove(vg_name: str):
+        if not vg_name:
+            raise ValueError("Name needed for VG remove operation.")
+
+        cmd = f"vgremove {vg_name}"
+        return TestRun.executor.run(cmd)
+
+    @staticmethod
+    def get_logical_volumes_path(vg_name: str):
+
+        cmd = f"lvdisplay | grep /dev/{vg_name}/ | awk '{{print $3}}'"
+        paths = TestRun.executor.run(cmd).stdout.splitlines()
+
+        return paths if paths else []
```

Comment:
splitlines() returns [] for empty stdout - no need to explicitly return [] if paths == []
```suggestion
        paths = TestRun.executor.run(cmd).stdout.splitlines()

        return paths
```